code/modules/mob/living/carbon/human/species_types/synths.dm

```diff
@@ -27,10 +27,12 @@
 	id = "military_synth"
 	armor = 25
 	punchdamage = 14
-	disguise_fail_health = 50
+	disguise_fail_health = 50 //This literally does nothing. This doesnt work.
 	changesource_flags = MIRROR_BADMIN | WABBAJACK
 
 /datum/species/synth/on_species_gain(mob/living/carbon/human/H, datum/species/old_species)
+	if (H.dna.features["legs"] == "Digitigrade Legs") //I fucking hate this.
+		initial_species_traits += DIGITIGRADE //If the target has digitigrade legs, store them for later
```

Comment:
this is handled in the parent call

Reply:
You'd think it would be fine, but it actually isnt. This needs to be here.

Reply:
(because i now realise you dont see the rest of the procs in this file)

Immediately after on_species_gain, all of your species vars are overwritten. Because DIGITIGRADE is an optional trait, not in the default Lizard trait list, it needs to be checked for immediately and stored for later, otherwise it goes fucky-wucky.